test/acceptance/helper.js

```diff
@@ -345,6 +344,185 @@ let suite = {
+    },
+    {
+      description: '@first in a loop with no items',
+      template: '{#names}<span class="{@first}first{/first}">{.}</span>{/names}',
+      context: {
+        names: []
+      },
+      expectedDom: (() => {
+        let frag = document.createDocumentFragment();
+        frag.appendChild(document.createTextNode(''));
+        return frag;
+      })()
+    },
+    {
+      description: '@first outside of a loop',
+      template: '<span class="{@first}first{/first}"></span>',
+      context: {
+        names: []
+      },
+      expectedDom: (() => {
+        let frag = document.createDocumentFragment();
+        let span = document.createElement('span');
+        span.setAttribute('class', '');
+        frag.appendChild(span);
+        return frag;
+      })()
+    },
+    {
+      description: '@last in a loop with one item',
+      template: '{#names}<span class="{@last}last{/last}">{.}</span>{/names}',
+      context: {
+        names: ['Steven']
+      },
+      expectedDom: (() => {
+        let frag = document.createDocumentFragment();
+        let span = document.createElement('span');
+        span.setAttribute('class', 'last');
+        span.appendChild(document.createTextNode('Steven'));
+        frag.appendChild(span);
+        return frag;
+      })()
+    },
+    {
+      description: '@last in a loop with multiple items',
+      template: '{#names}<span class="{@last}last{/last}">{.}</span>{/names}',
+      context: {
+        names: ['Steven', 'Jimmy', 'Prash']
+      },
+      expectedDom: (() => {
+        let frag = document.createDocumentFragment();
+        let span = document.createElement('span');
+        span.setAttribute('class', '');
+        span.appendChild(document.createTextNode('Steven'));
+        frag.appendChild(span);
+        span = document.createElement('span');
+        span.setAttribute('class', '');
+        span.appendChild(document.createTextNode('Jimmy'));
+        frag.appendChild(span);
+        span = document.createElement('span');
+        span.setAttribute('class', 'last');
+        span.appendChild(document.createTextNode('Prash'));
+        frag.appendChild(span);
+        return frag;
+      })()
+    },
+    {
+      description: '@last in a loop with no items',
+      template: '{#names}<span class="{@last}last{/last}">{.}</span>{/names}',
+      context: {
+        names: []
+      },
+      expectedDom: (() => {
+        let frag = document.createDocumentFragment();
+        frag.appendChild(document.createTextNode(''));
+        return frag;
+      })()
+    },
+    {
+      description: '@last outside of a loop',
+      template: '<span class="{@last}last{/last}"></span>',
+      context: {
+        names: []
+      },
+      expectedDom: (() => {
+        let frag = document.createDocumentFragment();
+        let span = document.createElement('span');
+        span.setAttribute('class', '');
+        frag.appendChild(span);
+        return frag;
+      })()
     }
```

Comment:
The nested loop tests are in. I'm not certain the new implementation of `helperContext` is the direction we want to go, but it's not bad. Here is a summary of how it works:

Each `section` (and `helper`) creates a new context item on the helper context stack (and removes that item from the stack when it's execution is complete. When you set a new value, (using `td.helpersContext#set`) the value is added to the topmost context item on the stack. When you get a value (using `td.helpersContext#get`) you search through each level of the stack until you find the value you are looking for. The bottom-most item on the stack is the "global" scope, which is where `get` searches last. You can put items onto the global scope using `td.helpersContext#setGlobal`. I didn't create a way to explicitly get items from the global scope, but I suppose that is possible.